### ISSUE TITLE
Deleting a comment on a non-contact post fails 

### ISSUE THREAD

**Issue by shmerl** [diaspora/diaspora#3666]
For some reason I can't delete my comment in the following post:

https://joindiaspora.com/posts/2019296

I.e. it disappears after delete is clicked, but it's not actually being deleted - as it shows up after refresh.

**Comment by Flaburgan** [comment]
We just reproduced that on framasphere.org, with a user of framasphere trying to delete his comment on a post by another user on framasphere too. So, the federation is not involved, and this is not specific to joindiaspora. I don't see anything in the logs, but there are only GET and POST requests, why isn't there any DELETE there?

**Comment by jhass** [comment]
Is the user savy enough to watch the network tab if its inspector?

**Comment by SansPseudoFix** [comment]
I tried to reproduce bug like the user mentionned by @Flaburgan but without succes, but I did it between framasphere and diaspora-fr. 
I post with my framasphere's account : I can delete my comments with this account.
I comment with my diaspora-fr's account. When I try to delete it  : 
* comment disapear if I look with this account
* comment **not deleted** if I look with my framaspere's account

I tried to check with browser's inspector (I'm not confortable with it, expect for HTML/CSS) : 
* sometimes I get ````Method : Delete````
* Sometimes... Nothing at all

Here a screenshot when "it works" (sorry, screenshot in french) : 

![delete_comments_delais](https://cloud.githubusercontent.com/assets/6507951/5909329/da536ca2-a5af-11e4-93e6-0951dd056309.png)

Left : when I try to delete with my diaspora-fr account.
Right : when I try to delete with my framasphere account (and when it works).

**Comment by Flaburgan** [comment]
@SansPseudoFix could you please make a test which doesn't involve the federation? Comment with a user on another user post and then try to delete the comment, with both users on the same pod?

**Comment by SansPseudoFix** [comment]
A user wanted to delete a comment on framasphere, and didn't success. So, I used podmin account to report this comment and delete it... But, it didn't work. Comment is still displayed, and I can't report it again. And no errors...

**Comment by Flaburgan** [comment]
The comment is still in the database.

**Comment by svbergerem** [comment]
@Flaburgan Could you try to delete the comment again and share the logfile?

**Comment by Flaburgan** [comment]
So we reproduce it again on framasphere. The author of [this comment](https://framasphere.org/posts/601455#cd0102d0c5730132ead92a0000053625) tries to delete it. There is a `DELETE` request which is sent to https://framasphere.org/comments/471845 (this link is a 404 btw... ?) and the comment disappears but is still there when the page is reloaded.

Here is what I found in the production.log
```
Started DELETE "/comments/471845" for 78.230.82.247 at 2015-04-16 10:48:32 +0200
Processing by CommentsController#destroy as JSON
  Parameters: {"id"=>"471845"}
{:event=>:sign_with_key, :status=>:complete}
event=push route=local sender=goffi@framasphere.org recipient=timyus_vegetari@framasphere.org payload_type=RelayableRetraction
  Rendered text template (0.0ms)
Completed 204 No Content in 19.2ms (Views: 0.5ms | ActiveRecord: 2.9ms)
user:8495 starting private receive from person:79a79d103c24013250b02a0000053625
event=receive status=abort reason='sender not connected to recipient' recipient=timyus_vegetari@framasphere.org sender=goffi@framasphere.org
error on receive: Diaspora::ContactRequiredUnlessRequest
```
So we have an answer: `status=abort reason='sender not connected to recipient'` it looks like the comment can't be delete because goffi doesn't share with timyus. I will ask togoffi if he shares with timyus, and if not, if he can start sharing with him and then try again to delete the comment.

**Comment by Flaburgan** [comment]
Goffi confirmed. He was not sharing with timyus, when he started sharing with him, he was able to remove the comment.

**Comment by jhass** [comment]
I'm not sure why that request is taking the private receiver route, either that is the bug or if it's supposed to work that way it's in the validations in https://github.com/diaspora/diaspora/blob/develop/lib/postzord/receiver/private.rb#L60

**Comment by Flaburgan** [comment]
Why did you tag that "federation", both users are on the same pod.

**Comment by jhass** [comment]
Because it errors out in the federation layer.

**Comment by svbergerem** [comment]
The same happens for likes. Both users are on the same pod. The user is able to like a post of the other use but he is not able to unlike the post.

```
INFO  PID-10073 TID-86612880 Rails: Started DELETE "/posts/5/likes/1" for ::1 at 2015-05-30 01:14:10 +0200
INFO  PID-10073 TID-86612880 Rails: <Hash> {:event=>:sign_with_key, :status=>:complete}
INFO  PID-10073 TID-86612880 Rails: event=push route=local sender=svbergerem@localhost:3000 recipient=testuser1@localhost:3000 payload_type=RelayableRetraction
INFO  PID-10073 TID-86612880 Rails: user:2 starting private receive from person:4ffcb2e0dea60132e102047d7b8eb256
ERROR PID-10073 TID-86612880 Rails: event=receive status=abort reason='sender not connected to recipient' recipient=testuser1@localhost:3000 sender=svbergerem@localhost:3000
INFO  PID-10073 TID-86612880 Rails: error on receive: Diaspora::ContactRequiredUnlessRequest
```

**Comment by denschub** [comment]
Closed by #6873.
